service/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/bean/ActivityServiceBeanTest.java

```diff
@@ -57,6 +60,244 @@
     @Mock
     private SpatialModuleService spatialModule;
 
+    @Test
+    public void getNextFishingActivity() {
+        // Given
+        int fishingActivityId = 5;
+        int expectedNextFishingActivityId = 18;
+
+        Instant now = Instant.now();
+
+        FishingTripEntity fishingTrip = new FishingTripEntity();
+
+        FishingActivityEntity first = new FishingActivityEntity();
+        first.setId(expectedNextFishingActivityId);
+        first.setOccurence(now.plus(5, ChronoUnit.MINUTES));
+
+        FishingActivityEntity second = new FishingActivityEntity();
+        second.setId(2);
+        second.setOccurence(now.minus(1, ChronoUnit.MINUTES));
+
+        FishingActivityEntity third = new FishingActivityEntity();
+        third.setId(fishingActivityId);
+        third.setOccurence(now);
+
+        FishingActivityEntity fourth = new FishingActivityEntity();
+        fourth.setId(10);
+        fourth.setOccurence(now.plus(55, ChronoUnit.MINUTES));
+
+        FishingActivityEntity fifth = new FishingActivityEntity();
+        fifth.setId(8);
+        fifth.setOccurence(now.minus(20, ChronoUnit.MINUTES));
+
+        fishingTrip.getFishingActivities().add(first);
+        fishingTrip.getFishingActivities().add(second);
+        fishingTrip.getFishingActivities().add(third);
+        fishingTrip.getFishingActivities().add(fourth);
+        fishingTrip.getFishingActivities().add(fifth);
+
+        third.setFishingTrip(fishingTrip);
+
+        when(fishingActivityDao.getFishingActivityById(fishingActivityId, null)).thenReturn(third);
+
+        // When
+        int nextFishingActivity = activityService.getNextFishingActivity(fishingActivityId);
+
+        // Then
+        assertEquals(expectedNextFishingActivityId, nextFishingActivity);
+    }
+
+    @Test
+    public void getNextFishingActivity_noNextActivity() {
+        // Given
+        int fishingActivityId = 5;
+
+        Instant now = Instant.now();
+
+        FishingTripEntity fishingTrip = new FishingTripEntity();
+
+        FishingActivityEntity fishingActivity = new FishingActivityEntity();
+        fishingActivity.setId(1);
+        fishingActivity.setOccurence(now.plus(5, ChronoUnit.MINUTES));
+
+        fishingTrip.getFishingActivities().add(fishingActivity);
+
+        fishingActivity.setFishingTrip(fishingTrip);
+
+        when(fishingActivityDao.getFishingActivityById(fishingActivityId, null)).thenReturn(fishingActivity);
+
+        // When
+        int nextFishingActivity = activityService.getNextFishingActivity(fishingActivityId);
+
+        // Then
+        assertEquals(-1, nextFishingActivity);
+    }
+
+    @Test
+    public void getNextFishingActivity_noActivity() {
+        // Given
+        int fishingActivityId = 5;
+
+        Instant now = Instant.now();
+
+        FishingTripEntity fishingTrip = new FishingTripEntity();
+
+        FishingActivityEntity fishingActivity = new FishingActivityEntity();
```

Comment:
Is there a point to setting up this fishing activity? (I mean the test is named "no activity")